### ISSUE TITLE
Up/Down events not firing as expected when other controls are present in the page layout

### ISSUE THREAD

**Issue by MitchBomcanhao** [MichaelRumpler/GestureSample#47]
I've been trying MR.Gestures for the first time and found something quite odd.
I'm running a MAUI app on a windows pc, with a mouse, and I noticed that the events/commands were not being fired as I expected, resulting in a very inconsistent user experience.
For example, using different mouse buttons I will get different events firing at different rates, eg, the middle mouse button worked pretty well, but the left and right buttons had very sketchy results. 

Here's an example. 

1. load the app (I just made a blank maui app, added the MR.Gestures package, used the ui stuff shown below), 
2. click 3 times in one item with the left mouse button, 
3. click it with the middle mouse button
4. move the cursor away from the app
5. OBSERVE in the debugger output window:

```
## event name
## list of all the TouchSource values in the event args
```

```
(1st click)
## pressing down: 
## via LeftMouseButton

(2nd click)
## pressing down:
## via LeftMouseButton
## via LeftMouseButton

(3rd click)
## pressing down:
## via LeftMouseButton
## via LeftMouseButton
## via LeftMouseButton

(Click with the middle mouse button)
## pressing down:
## via LeftMouseButton
## via LeftMouseButton
## via LeftMouseButton
## via MiddleMouseButton
## pressing up:
## via LeftMouseButton
## via LeftMouseButton
## via LeftMouseButton
## via MiddleMouseButton

(moving the mouse pointer away from the app)
## pressing up:
## via MousePointer
## via LeftMouseButton
## via MiddleMouseButton
## pressing up:
## via LeftMouseButton
## via MiddleMouseButton
## pressing up:
## via MiddleMouseButton
```

That is very different from what I expected to happen, which would be a pair of DOWN/UP events for each click, with a single touch source each. 
If I remove the Button from my page's xaml, all works as it appears it should:
```
## pressing down:
## via LeftMouseButton
## pressing up:
## via LeftMouseButton

## pressing down:
## via LeftMouseButton
## pressing up:
## via LeftMouseButton

## pressing down:
## via LeftMouseButton
## pressing up:
## via LeftMouseButton

## pressing down:
## via MiddleMouseButton
## pressing up:
## via MiddleMouseButton
```

 It is very likely something similarly fishy is going on with commands too.

this is the UI code:
```
    <ScrollView>
        <VerticalStackLayout Spacing="0">
            <VerticalStackLayout
                x:Name="ItemsContainer"
                Padding="30,0"
                BindableLayout.ItemsSource="{Binding}"
                Spacing="25">
                <BindableLayout.ItemTemplate>
                    <DataTemplate>
                        <mr:StackLayout
                            Background="LightBlue"
                            Down="ContentView_Down"
                            Up="ContentView_Up">
                            <Label Text="Click here" />
                        </mr:StackLayout>
                    </DataTemplate>
                </BindableLayout.ItemTemplate>
            </VerticalStackLayout>
            <Button Text="this button existing breaks the up/down events" />
        </VerticalStackLayout>
    </ScrollView>
```
and here is the code behind
```
    public partial class MainPage : ContentPage
    {
        public MainPage()
        {
            InitializeComponent();

            this.ItemsContainer.BindingContext = new List<int>() { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
        }

        private void ContentView_Down(object sender, MR.Gestures.DownUpEventArgs e)
        {
            System.Diagnostics.Debug.WriteLine("\n");
            System.Diagnostics.Debug.WriteLine("## pressing down:");
            foreach (var source in e.Sources)
            {
                System.Diagnostics.Debug.WriteLine($"## via {source}");
            }
        }

        private void ContentView_Up(object sender, MR.Gestures.DownUpEventArgs e)
        {
            System.Diagnostics.Debug.WriteLine("## pressing up:");
            foreach (var source in e.Sources)
            {
                System.Diagnostics.Debug.WriteLine($"## via {source}");
            }
        }
    }
```

**Comment by MichaelRumpler** [comment]
Thanks! I [added your code to the GestureSample](https://github.com/MichaelRumpler/GestureSample/commit/c136186258ccd14512b9bc241057e9b6c34944f3) and it is reproducible.
Strange that that `Button` causes it. I'll have to debug it.

**Comment by MichaelRumpler** [comment]
I found a fix for that.

But I also saw that MAUI does not raise the `HandlerChanging` event for the elements in the template. Therefore the cleanup doesn't work. I'll see if I can work around that somehow.

**Comment by MichaelRumpler** [comment]
It turned out that is a bug in MAUI. It also does not cleanup a page with your content when you use the standard MAUI controls instead of the MR.Gestures ones.

I released version 4.0.1 which should fix this issue. Please close this issue if it fixed the problem for you.
I'll open a issue in MAUI next week to fix the memory leak.

**Comment by MitchBomcanhao** [comment]
Thanks for looking into the issue.
First impressions are that the down/up events are now firing as expected, but this is but a cursory check on the issue as described (which I'm sure is less than what you did in your own testing). If I do find further problems, I'll try to post them here, but at the moment my time to explore the gestures package is limited.

it is not surprising that there are underlying issues from MAUI, though, my experience with migrating a xamarin forms app to MAUI shows that a lot of it is very broken indeed.
